pkg/cleanup/containers.go

```diff
@@ -134,19 +134,19 @@ func (c *containers) stopContainerd() {
 
 func (c *containers) stopAllContainers() error {
 	var errs []error
-	logrus.Debugf("trying to list all pods")
 
 	var pods []string
 	err := retry.Do(func() error {
+		logrus.Debugf("trying to list all pods")
 		var err error
 		pods, err = c.Config.containerRuntime.ListContainers()
 		if err != nil {
 			return err
 		}
 		return nil
-	})
+	}, retry.Attempts(3))
 	if err != nil {
-		logrus.Debugf("failed at listing pods %v", err)
+		logrus.Errorf("failed at listing pods %v", err)
 		return err
```

Comment:
What about  removing that log completely, since the err gets returned to the caller anyways?

```suggestion
		return fmt.Errorf("failed at listing pods %w", err)
```